File: app/src/main/java/com/example/saka/myapplication/model/Post.java
package com.example.saka.myapplication.model;

import java.util.HashMap;
import java.util.Map;

public class Post {

    private String uid;
    private String author;
    private String title;
    private String body;
    private int starCount = 0;
    private Map<String, Boolean> stars = new HashMap<>();

    public Post() {
        //  Default constructor
    }

    public Post(String uid, String author, String title, String body) {
        this.uid = uid;
        this.author = author;
        this.title = title;
        this.body = body;
    }

    public Map<String, Object> toMap() {
        Map<String, Object> map = new HashMap<>();
        map.put("uid", uid);
        map.put("author", author);
        map.put("title", title);
        map.put("body", body);
        map.put("starCount", starCount);
        //map.put("stars", stars);
        return map;
    }

    public String getUid() {
        return uid;
    }

    public void setUid(String uid) {
        this.uid = uid;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }

    public int getStarCount() {
        return starCount;
    }

    public void setStarCount(int starCount) {
        this.starCount = starCount;
    }

    public Map<String, Boolean> getStars() {
        return stars;
    }

    public void setStars(Map<String, Boolean> stars) {
        this.stars = stars;
    }
}

File: app/src/main/java/com/example/saka/myapplication/model/Post.kt
package com.example.saka.myapplication.model

class Post {

    var uid: String? = null
    var author: String? = null
    var title: String? = null
    var body: String? = null
    var starCount = 0
    var stars: MutableMap<String, Boolean> = HashMap()

    constructor() {
        //  Default constructor
    }

    constructor(uid: String?, author: String?, title: String?, body: String?) {
        this.uid = uid
        this.author = author
        this.title = title
        this.body = body
    }

    fun toMap(): Map<String, Any?> {
        val map: MutableMap<String, Any?> = HashMap()
        map["uid"] = uid
        map["author"] = author
        map["title"] = title
        map["body"] = body
        map["starCount"] = starCount
        // map.put("stars", stars);
        return map
    }
}
